src/controllers/currentWarningsController.js

```diff
@@ -28,6 +28,15 @@ const currentWarningsController = function (currentWarnings) {
   };
 
   const postNewWarningDescription = async (req, res) => {
+    const userRole = req.body?.requestor?.role;
+    const isPrivilegedUser = userRole === 'Owner' || userRole === 'Administrator';
+    if (!isPrivilegedUser &&
+      !(await helper.hasPermission(req.body.requestor, 'addWarningTracker')) 
+      
+    ) {
```

Comment:
```suggestion
    if (!(await helper.hasPermission(req.body.requestor, 'addWarningTracker')) 
) {
```
`hasPermission()` checks role permissions as well, so `isPrivilegedUser` should be redundant. You can add `'addWarningTracker',` (and all other relevant permissions) to the Admin and Owner sections of `createInitialPermissions.js` if you want to guarantee that those roles always have this permission. You should at least do that for Owner.